client/matching/loadbalancer.go

```diff
@@ -134,3 +142,34 @@ func (lb *defaultLoadBalancer) pickPartition(
 
 	return fmt.Sprintf("%v%v/%v", taskListPartitionPrefix, taskList.GetName(), p)
 }
+
+func (lb *defaultLoadBalancer) GetAllPartitions(
+	domainID string,
+	taskList shared.TaskList,
+	taskListType int,
+) ([]string, error) {
+	domainName, err := lb.domainIDToName(domainID)
+	var partitionKeys []string
+	if err != nil {
+		return partitionKeys, err
+	}
+
+	taskListName := taskList.GetName()
+	if strings.HasPrefix(taskListName, taskListPartitionPrefix) {
```

Comment:
I believe this is used in more than one place in this file, can you separate this logic into its own helper method for clarity ?